app/Models/Traits/HasThresholds.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * HasLimits.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ *
+ * @copyright  2025 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace App\Models\Traits;
+
+use LibreNMS\Enum\Severity;
+
+trait HasThresholds
+{
+    public function currentStatus(): Severity
+    {
+        if ($this->sensor_class == 'state') {
+            return $this->currentTranslation()?->severity() ?? Severity::Unknown;
+        }
+
+        if ($this->sensor_current === null) {
+            return Severity::Unknown;
+        }
+
+        if ($this->sensor_limit !== null && $this->sensor_current >= $this->sensor_limit) {
```

Comment:
Aren’t these saved with a mix of null and 0 ?

Reply:
yes, the limit can be 0 or unset.

Reply:
FYI, this code was unchanged, just moved to a the trait.